src/clj/clojure/core.clj

```diff
@@ -4037,7 +4037,7 @@
    :static     true
    :deprecated "1.9"}
   [ns]
-  (.getName (the-ns ns)))
+  (symbol (.getName (the-ns ns))))
```

Comment:
I wanted Namespace and Var to participate in Named, because they do have names and we have an interface for that so it makes sense and I just got a Java method name conflict.

In retrospect I shouldn't have merged that change without thinking about it more and trying to analyze uses. Amusingly, nothing in core broke because of the type chance, since the uses were all `(name (ns-name (the-ns ..)))`, so `(name String) -> String` and `(ns-name Ns) -> String)` same result type.

Deprecating `ns-name` was also probably a mistake in retrospect, since although `name` is mostly a drop-in replacement it does have a unique purpose. Maybe there's call here for a `sym-name` or something that's generic to Named. Need to think rather than just merging.